examples/images_contours_and_fields/plot_streamplot.py

```diff
@@ -22,45 +22,45 @@
 V = 1 + X - Y**2
 speed = np.sqrt(U**2 + V**2)
 
-fig = plt.figure(figsize=(7, 9))
-gs = gridspec.GridSpec(nrows=3, ncols=2, height_ratios=[1, 1, 2])
+
+gs = {'height_ratios': [1, 1, 2]}
+fig, axs = plt.subplots(nrows=3, ncols=2, gridspec_kw=gs, figsize=(7, 9))
 
 #  Varying density along a streamline
-ax0 = fig.add_subplot(gs[0, 0])
-ax0.streamplot(X, Y, U, V, density=[0.5, 1])
-ax0.set_title('Varying Density')
+axs[0, 0].streamplot(X, Y, U, V, density=[0.5, 1])
+axs[0, 0].set_title('Varying Density')
 
 # Varying color along a streamline
-ax1 = fig.add_subplot(gs[0, 1])
-strm = ax1.streamplot(X, Y, U, V, color=U, linewidth=2, cmap='autumn')
-fig.colorbar(strm.lines)
-ax1.set_title('Varying Color')
+strm = axs[0, 1].streamplot(X, Y, U, V, color=U, linewidth=2, cmap='autumn')
+fig.colorbar(strm.lines, ax=axs[0, 1])
+axs[0, 1].set_title('Varying Color')
 
 #  Varying line width along a streamline
-ax2 = fig.add_subplot(gs[1, 0])
 lw = 5*speed / speed.max()
-ax2.streamplot(X, Y, U, V, density=0.6, color='k', linewidth=lw)
-ax2.set_title('Varying Line Width')
+axs[1, 0].streamplot(X, Y, U, V, density=0.6, color='k', linewidth=lw)
+axs[1, 0].set_title('Varying Line Width')
 
 # Controlling the starting points of the streamlines
 seed_points = np.array([[-2, -1, 0, 1, 2, -1], [-2, -1,  0, 1, 2, 2]])
 
-ax3 = fig.add_subplot(gs[1, 1])
-strm = ax3.streamplot(X, Y, U, V, color=U, linewidth=2,
-                      cmap='autumn', start_points=seed_points.T)
-fig.colorbar(strm.lines)
-ax3.set_title('Controlling Starting Points')
+strm = axs[1, 1].streamplot(X, Y, U, V, color=U, linewidth=2,
+                            cmap='autumn', start_points=seed_points.T)
+fig.colorbar(strm.lines, ax=axs[1, 1])
+axs[1, 1].set_title('Controlling Starting Points')
 
 # Displaying the starting points with blue symbols.
-ax3.plot(seed_points[0], seed_points[1], 'bo')
-ax3.set(xlim=(-w, w), ylim=(-w, w))
+axs[1, 1].plot(seed_points[0], seed_points[1], 'bo')
+axs[1, 1].set(xlim=(-w, w), ylim=(-w, w))
 
 # Create a mask
 mask = np.zeros(U.shape, dtype=bool)
 mask[40:60, 40:60] = True
 U[:20, :20] = np.nan
 U = np.ma.array(U, mask=mask)
 
+gs = axs[0, 0].get_subplotspec().get_gridspec()
+axs[2, 0].remove()
+axs[2, 1].remove()
```

Comment:
Again, I don't think we should use example with `ax.remove()` in them.  I'd prefer we just added each of these (i.e. the old behaviour was fine).  But maybe other folks have other opinions.

Reply:
Hi,
I agree it was a bit messy to use the subplots method here. I reverted the file to the original version, as you recommended.
How are the other files? 
Also, I cannot figure out why the codecov test does not pass sometimes.
Thanks